Sources/Ink/Internal/Character+Escaping.swift

```diff
@@ -3,8 +3,80 @@
 *  Copyright (c) John Sundell 2019
 *  MIT license, see LICENSE file for details
 */
+// ASCII chars escapable from CommonMark spec
+// \!\"\#\$\%\&\'\(\)\*\+\,\-\.\/\:\;\<\=\>\?\@\[\\\]\^\_\`\{\|\}\~
+// Result desired
+// !&quot;#$%&amp;'()*+,-./:;&lt;=&gt;?@[\]^_`{|}~
+// An ASCII punctuation character is !, ", #, $, %, &, ', (, ), *, +, ,, -, .,/
+// (U+0021–2F), :, ;, <, =, >, ?, @ (U+003A–0040), [, \, ], ^, _,`
+// (U+005B–0060), {, |, }, or ~ (U+007B–007E).
+let escapeSubstitutions: [Character: String] = [
+    "!": "!",
```

Comment:
Why do we need entries for characters that are not being substituted, where the key and value are the exact same (which seems to be the majority of entries)? 🤔

Reply:
This is not obvious and certainly deserves better comment to explain.
The amazing spec is that escaped CHARACTERS are not escaped (look at following test case, testOtherCharactersNotEscaped) The easiest way to make that work is to exhaustively handle ALL the specified punctuation chars at this dict lookup.

Reply:
I cleaned it up a little as I misunderstood the spec. The Unicode hex stuff was not extra characters; just an explanation of the preceding ones. They should have used a semicolon since it was already comma separated.

Reply:
After further playing with code blocks testing, I had to reintroduce the original HTML only escaping for use inside code and the punctuation testing escape for other cases.